### PR TITLE
AmplitudeTap: Peak Analysis Mode

### PR DESCRIPTION
**Pull request by Matt54**
Created AnalysisMode enum to determine if the returned amplitude value is rms or peak

### CODE REVIEW

File: Sources/AudioKit/Analysis/AmplitudeTap.swift [hunk 1]
```diff
@@ -23,15 +23,20 @@ public class AmplitudeTap: BaseTap {
...
     private var handler: (Float) -> Void = { _ in }
 
     /// Initialize the amplitude
     ///
     /// - parameter input: Node to analyze
     /// - parameter bufferSize: Size of buffer to analyze
     /// - parameter handler: Code to call with new amplitudes
-    public init(_ input: Node, bufferSize: UInt32 = 1_024, handler: @escaping (Float) -> Void = { _ in }) {
+    /// - parameter mode: rms or peak returned amplitudes
+    public init(_ input: Node, bufferSize: UInt32 = 1_024, mode: AnalysisMode = .rms, handler: @escaping (Float) -> Void = { _ in }) {
```
[the remark this answers is not in the document]
Comment (Matt54) [reply Sources/AudioKit/Analysis/AmplitudeTap.swift:37]: How do we feel about long initializers? Should each parameter go on a separate line?

[the remark this answers is not in the document]
Comment (aure) [reply Sources/AudioKit/Analysis/AmplitudeTap.swift:37]: Yeah, each on its own line once you are past 120 characters if they all were on one line.

**Comment by aure** [comment]
Modes deserve tests.

**Comment by aure** [comment]
Cool!

**Pull request merged**